### PR TITLE
Refactoring: extract methods from `ESRestTestCase#initClient`

### PR DESCRIPTION
**Pull request by ldematte**
Initialization of information about the cluster under test is factored out into 2 methods; each method is independent (does not rely on state, static or not) so it can be used with different admin clients (e.g. in case of multi-cluster, like CCS).

First step to solve the potential issue identified in https://github.com/elastic/elasticsearch/pull/103560
Once this is merged, we'll need to update https://github.com/elastic/elasticsearch/pull/103311 to use the function(s) introduced here.

### CODE REVIEW

**Review by thecoop — APPROVED** [review]
Looks sensible

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 1]
```diff
@@ -216,9 +216,27 @@ public enum ProductFeature {
     }
 
     private static EnumSet<ProductFeature> availableFeatures;
-    private static Set<String> nodeVersions;
+    private static Set<String> nodesVersions;
     private static TestFeatureService testFeatureService;
 
+    protected static Set<String> getCachedNodeVersions() {
```
Comment (mosche) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java:222]: This should be plural (Node**s** ) as well. 
Nit, how about just using `nodesVersions` to not make this look like a regular getter?

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 2]
```diff
@@ -233,24 +251,20 @@ public void initClient() throws IOException {
...
             assert testFeatureService == null;
             clusterHosts = parseClusterHosts(getTestRestCluster());
             logger.info("initializing REST clients against {}", clusterHosts);
             client = buildClient(restClientSettings(), clusterHosts.toArray(new HttpHost[clusterHosts.size()]));
             adminClient = buildClient(restAdminSettings(), clusterHosts.toArray(new HttpHost[clusterHosts.size()]));
 
             availableFeatures = EnumSet.of(ProductFeature.LEGACY_TEMPLATES);
-            nodeVersions = new TreeSet<>();
-            var semanticNodeVersions = new HashSet<Version>();
+            nodesVersions = new TreeSet<>();
```
Comment (mosche) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java:262]: How about wrapping this into an unmodifiable set, especially if getting exposed now?

**Review by mosche — COMMENTED** [review]
Looking good, just some minor comments

**Pull request merged**